ir/inst_other.go

```diff
@@ -674,3 +674,48 @@ func (inst *InstCleanupPad) LLString() string {
 	}
 	return buf.String()
 }
+
+// ~~~ [ freeze ] ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+// InstFreeze is an LLVM IR freeze instruction.
+type InstFreeze struct {
+	// Name of local variable associated with the result.
+	LocalIdent
+
+	X value.Value
+
+	// extra.
+
```

Comment:
I think we need to add a `Typ` field here, unfortunately, for type caching. This is to be able handle types for recursive and post-declared local identifiers, e.g. using `phi` instructions and such.

```
	// Type of result produced by the instruction.
	Typ types.Type
```

Reply:
Sorry. Is the freeze instruction always the `token` type? I thought it would be the type of `x`.

Reply:
It would be type of X, my fault. So do we still need this `Typ` field?